borsh-derive-internal/src/struct_ser.rs

```diff
@@ -16,6 +16,10 @@ pub fn struct_ser(input: &ItemStruct, cratename: Ident) -> syn::Result<TokenStre
         },
         Clone::clone,
     );
+    let predicates = compute_predicates(&input.generics, &cratename);
+    predicates
+        .into_iter()
+        .for_each(|predicate| where_clause.predicates.push(predicate));
```

Comment:
Looks like a job for [`extend()`](https://docs.rs/syn/latest/syn/punctuated/struct.Punctuated.html#method.extend)

```suggestion
    where_clause.predicates.extend(predicates);
```